src/hooks/v2/contractReader/V2ContractReader.ts

```diff
@@ -0,0 +1,150 @@
+import { Contract, EventFilter } from '@ethersproject/contracts'
+import { V2UserContext } from 'contexts/v2/userContext'
+
+import { V2ContractName, V2Contracts } from 'models/v2/contracts'
+import { useCallback, useContext, useState } from 'react'
+import { useDeepCompareEffectNoCheck } from 'use-deep-compare-effect'
+
+export type ContractUpdateOn = {
+  contract?: ContractConfig
+  eventName?: string
+  topics?: EventFilter['topics']
+}[]
+
+export type ContractConfig = V2ContractName | Contract | undefined
+
+export default function useV2ContractReader<V>({
```

Comment:
We might be able to create a generic `useContractReader` honestly, but this was the path of least resistance and doesn't feel that consequential.